addons/mail/static/src/models/thread/thread.js

```diff
@@ -1385,8 +1388,12 @@ function factory(dependencies) {
             if (this.channel_type === 'chat' && this.correspondent) {
                 return this.custom_channel_name || this.correspondent.nameOrDisplayName;
             }
-            if (this.channel_type === 'group') {
-                return this.name || this.members.map(partner => partner.nameOrDisplayName).join(', ');
+            if (this.channel_type === 'group' && !this.name) {
+                let groupName = this.members.map(partner => partner.nameOrDisplayName);
+                if (this.guestMembers.length) {
+                    groupName = groupName.concat(this.guestMembers.map(guest => guest.name));
+                }
+                return groupName.sort().join(', ');
```

Comment:
After discussing this with the PO, we should sort members as follows: alphabeticaly sorted partners, alphabeticaly sorted guests.